pkg/core/mempool/mempool.go

```diff
@@ -47,14 +47,29 @@ func NewMempool(logger *common.Logger, config *config.Config, db *db.Queries, ma
 		config:                 config,
 		deque:                  list.New(),
 		txMap:                  make(map[string]*list.Element),
-		broadcastPeers:         make(map[*sdk.Sdk]struct{}),
+		broadcastPeers:         make(map[string]*sdk.Sdk),
 		db:                     db,
 		maxMempoolTransactions: maxTransactions,
 	}
 }
 
-func (m *Mempool) AddPeer(peer *sdk.Sdk) {
-	m.broadcastPeers[peer] = struct{}{}
+func (m *Mempool) AddPeer(address, endpoint string) error {
+	parsedURL, err := url.Parse(endpoint)
+	if err != nil {
+		m.logger.Errorf("could not parse url for %s: %v", endpoint, err)
+		return err
+	}
+
+	oapiendpoint := parsedURL.Host
+	peerSdk, err := sdk.NewSdk(sdk.WithOapiendpoint(oapiendpoint))
+	if err != nil {
+		m.logger.Errorf("could not init sdk for %s %s: %v", oapiendpoint, address, err)
+		return err
+	}
+
+	m.broadcastPeers[address] = peerSdk
```

Comment:
i don't think this will cause many problems but we do have some nodes with matching addresses so we would have a missing peer at times